src/getLocationForJsonPath.ts

```diff
@@ -66,16 +70,22 @@ function getEndPosition(node: YAMLNode): number {
   return node.endPosition;
 }
 
-function findNodeAtPath(node: YAMLNode, path: JsonPath, closest: boolean) {
+function findNodeAtPath(
+  node: YAMLNode,
+  path: JsonPath,
+  { closest, mergeKeys }: { closest: boolean; mergeKeys: boolean }
+) {
   pathLoop: for (const segment of path) {
     switch (node && node.kind) {
       case Kind.MAP:
-        for (const item of node.mappings) {
+        const mappings = getMappings(node.mappings, mergeKeys);
+        for (let i = mappings.length - 1; i >= 0; i--) {
```

Comment:
Curious why you're going through the mappings from the last to the first

Reply:
Will add a comment in the code.